plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -1234,6 +1246,15 @@ public Answer createTemplateFromVolume(CopyCommand cmd) {
 
             details = "create template from volume exception: " + VmwareHelper.getExceptionMessage(e);
             return new CopyCmdAnswer(details);
+        } finally {
+            try {
+                if (volume.getVmName() == null && workerVmMo != null) {
+                    workerVmMo.detachAllDisks();
+                    workerVmMo.destroy();
+                }
+            } catch (Throwable e) {
+                s_logger.warn("Failed to destroy worker VM created for detached volume");
```

Comment:
@Pearl1594 will it be better to log the exception here as it could be during detach operation as well?

Reply:
@shwstppr I could change it to an error log, however, I'm not sure if an exception is required as the preliminary job of creating the template is successful, and the cleanup has some issue.